packages/dashboard/src/dashboard-actions/dashboardReducer.spec.ts

```diff
@@ -1,17 +1,283 @@
+import { DashboardReducerState, DashboardConfiguration } from '../types';
+import { onDeleteAction, onCopyAction, onEditCellSizeAction, onEditWidthAction } from '../dashboard-actions/actions';
+import {
+  MOCK_EMPTY_DASHBOARD,
+  MOCK_KPI_WIDGET,
+  MOCK_LINE_CHART_WIDGET,
+  MockDashboardFactory,
+  MockWidgetFactory,
+  dashboardConfig,
+} from '../testing/mocks';
+import { move } from './move';
+import { deleteWidgets } from './delete';
+import { trimWidgetPosition } from '../components/iot-dashboard/trimWidgetPosition';
 
-it('returns empty dashboard configuration when provided an empty dashboard', () => {
-  expect(
-    dashboardReducer(
-      MOCK_EMPTY_DASHBOARD,
+describe('MOVE', () => {
+  it('returns empty dashboard configuration when provided an empty dashboard', () => {
+    expect(
+      dashboardReducer(
+        {
+          dashboardConfiguration: MOCK_EMPTY_DASHBOARD,
+          intermediateDashboardConfiguration: undefined,
+          selectedWidgetIds: [],
+          numTimesCopyGroupHasBeenPasted: 0,
+          copyGroup: [],
+          stretchToFit: true,
+          width: 1000,
+          cellSize: 30,
+        },
+        onMoveAction({
+          widgetIds: [],
+          cellSize: 10,
+          position: { x: 10, y: 10 },
+          prevPosition: { x: 20, y: 10 },
+          isActionComplete: true,
+        })
+      )
+    ).toEqual({
+      dashboardConfiguration: MOCK_EMPTY_DASHBOARD,
+      intermediateDashboardConfiguration: undefined,
+      selectedWidgetIds: [],
+      numTimesCopyGroupHasBeenPasted: 0,
+      copyGroup: [],
+      stretchToFit: true,
+      width: 1000,
+      cellSize: 30,
+    });
+  });
+  it('moves a single widget', () => {
+    const state: DashboardReducerState = {
+      dashboardConfiguration: {
+        widgets: [MOCK_KPI_WIDGET],
+        viewport: { duration: '5m' },
+      },
+      selectedWidgetIds: [MOCK_KPI_WIDGET.id],
+      numTimesCopyGroupHasBeenPasted: 0,
+      copyGroup: [],
+      stretchToFit: true,
+      width: 1000,
+      cellSize: 30,
+      intermediateDashboardConfiguration: undefined,
+    };
+    const movedDashboard: DashboardReducerState = state;
+    movedDashboard.dashboardConfiguration = move({
+      selectedWidgetIds: [MOCK_KPI_WIDGET.id],
+      cellSize: 10,
+      position: { x: 10, y: 10 },
+      previousPosition: { x: 20, y: 20 },
+      dashboardConfiguration: state.dashboardConfiguration,
+    });
+    movedDashboard.dashboardConfiguration.widgets.map(trimWidgetPosition);
+
+    const reducerDashboard = dashboardReducer(
+      state,
       onMoveAction({
-        widgetIds: [],
+        widgetIds: [MOCK_KPI_WIDGET.id],
         cellSize: 10,
         position: { x: 10, y: 10 },
-        prevPosition: { x: 14, y: 10 },
+        prevPosition: { x: 20, y: 20 },
+        isActionComplete: true,
       })
-    )
-  ).toEqual(MOCK_EMPTY_DASHBOARD);
+    ).dashboardConfiguration;
+    reducerDashboard.widgets.map(trimWidgetPosition);
```

Comment:
is there some way to have everything structured in a way that we don't need to a complex process before asserting in the test?